[Instrumentation.GrpcNetClient] Support only stable HTTP semantic convention (#5259)
open-telemetry/opentelemetry-dotnet: src/OpenTelemetry.Instrumentation.GrpcNetClient/CHANGELOG.md

  [issue](https://github.com/open-telemetry/opentelemetry-dotnet/issues/5092)
  for details and workaround.
  ([#5077](https://github.com/open-telemetry/opentelemetry-dotnet/pull/5077))

> * Removed `OTEL_SEMCONV_STABILITY_OPT_IN` environment variable support. The

Reviewer: One question .. The grpc conventions are still not stable and this instrumentaion is still emitting them. Should we explicitly call it out, to avoid giving the feel that the entire conventions emitted by this is stable?

Reviewer 2: I think it's fine, folks who want to spend time reading changelog should already know semver2.

> * Removed support for `http` and `http/dup` values for`OTEL_SEMCONV_STABILITY_OPT_IN`

Reviewer: `OTEL_SEMCONV_STABILITY_OPT_IN` - this entire env variable is now ignored right? Maybe make it explicitly clear that this env var will be ignored now.

> * Removed support for `http` and `http/dup` values for`OTEL_SEMCONV_STABILITY_OPT_IN`

Reviewer 2: ```suggestion
* Removed support for `http` and `http/dup` values for `OTEL_SEMCONV_STABILITY_OPT_IN`
```

> * Removed support for `http` and `http/dup` values for`OTEL_SEMCONV_STABILITY_OPT_IN`
>   environment variable. The library will now emit only the
>   [stable](https://github.com/open-telemetry/semantic-conventions/tree/v1.23.0/docs/http)
>   HTTP semantic conventions.

Reviewer 2: ```suggestion
* Removed `OTEL_SEMCONV_STABILITY_OPT_IN` environment variable support. The
  library will now emit the
  [stable](https://github.com/open-telemetry/semantic-conventions/tree/v1.23.0/docs/http)
  HTTP semantic conventions.
```

> * Removed `OTEL_SEMCONV_STABILITY_OPT_IN` environment variable support. The
>   library will now emit the
>   [stable](https://github.com/open-telemetry/semantic-conventions/tree/v1.23.0/docs/http)
>   HTTP semantic conventions.
>   ([#5259](https://github.com/open-telemetry/opentelemetry-dotnet/pull/5259))

Reviewer 3: Sorry, chiming in with more nits with regards to the changelog 😆. This instrumentation does not really use the HTTP conventions but rather the shared network conventions that are now also stable.

```suggestion
* Removed support for the `OTEL_SEMCONV_STABILITY_OPT_IN` environment variable
  which toggled the use of the new conventions for the
  [server, client, and shared network attributes](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/attributes.md#server-client-and-shared-network-attributes).
  Now that this suite of attributes are stable, this instrumentation will only
  emit the new attributes.
  ([#5259](https://github.com/open-telemetry/opentelemetry-dotnet/pull/5259))
```
